### PR TITLE
Replaces the M41A1 with the M412

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Removes the M41A1 from the game entirely, replacing it with a different gun that's more in style with the servers look in general, and with different lore rather than snowflake random things.
Need to get an ammo sprite for a few more different things though.
Otherwise should be done?
The M412 has 5% more damage than the AR (which its most comparable to) because it has 20%ish less ammo than it.

## Why It's Good For The Game
Adds guns that fit our gun style more.

## Changelog
:cl:
add: Added new M412 series of rifles.
del: Removed old Pulse rifles
fix:Some indentation issues that the HPR had
balance: M412 has 5% more damage than the AR, M412L1 (the HPR) has slightly better fire delay. M412 families stat is in general consistent with the normal AR family.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -49. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

File: _maps/map_files/Pillar_of_Spring/TGS_Pillar_of_Spring.dmm [hunk 1]
```diff
@@ -757,7 +757,7 @@
 /obj/machinery/camera/autoname/mainship{
 	dir = 4
 	},
-/obj/item/weapon/gun/rifle/m41a1{
+/obj/item/weapon/gun/rifle/m412{
 	desc = "An outdated rifle for the TerraGov Marine Corps, carried by a few coporate mercenary groups, the M41A1 is a very rare sight in TerraGov systems. Uses 10x24mm caseless ammunition. Inscribed is Moth"
```
Comment (XSlayer300) [inline _maps/map_files/Pillar_of_Spring/TGS_Pillar_of_Spring.dmm:761]:
```suggestion
	desc = "The M412 rifle is a Pulse Industries rifle, billed as a pulse rifle due to its use of electronic firing for faster velocity. A rather common sight in most systems. Uses 10x24mm caseless ammunition. Inscribed is a moth stencil and it looks worn."
```

File: code/game/objects/structures/crates_lockers/largecrate_supplies.dm [hunk 2]
```diff
@@ -143,12 +143,12 @@
 	icon_state = "case"
 
 /obj/structure/largecrate/supply/ammo/m41a
-	name = "\improper M41A1 magazine case (x20)"
+	name = "\improper M412 magazine case (x20)"
 	desc = "An ammunition case containing 20 M41A1 magazines."
```
Comment (XSlayer300) [inline code/game/objects/structures/crates_lockers/largecrate_supplies.dm:147]:
```suggestion
	desc = "An ammunition case containing 20 M412 magazines."
```

File: code/game/objects/structures/crates_lockers/largecrate_supplies.dm [hunk 3]
```diff
@@ -143,12 +143,12 @@
...
 	supplies = list(/obj/item/ammo_magazine/rifle = 20)
 
 /obj/structure/largecrate/supply/ammo/m41a_box
-	name = "\improper M41A1 ammunition box case (x4)"
+	name = "\improper M412 ammunition box case (x4)"
 	desc = "An ammunition case containing four M41A1 600 round boxes of ammunition."
```
Comment (XSlayer300) [inline code/game/objects/structures/crates_lockers/largecrate_supplies.dm:152]:
```suggestion
	desc = "An ammunition case containing four M412 600 round boxes of ammunition."
```

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -854,10 +854,10 @@ Defined in conflicts.dm of the #defines folder.
 
 
 /obj/item/attachable/stock/rifle
-	name = "\improper M41A1 skeleton stock"
-	desc = "A rare stock distributed in small numbers to TGMC forces. Compatible with the M41A1, this stock reduces recoil and improves accuracy, but at a reduction to handling and agility. Seemingly a bit more effective in a brawl."
+	name = "\improper M412 solid stock"
+	desc = "A common stock used by the M412 pulse rifle series, used by long rifles this stock reduces recoil and improves accuracy, but at a reduction to handling and agility. Seemingly a bit more effective in a brawl."
```
Comment (XSlayer300) [inline code/modules/projectiles/gun_attachables.dm:858]:
```suggestion
	desc = "A common stock used by the M412 pulse rifle series, used for long rifles. This stock reduces recoil and improves accuracy, but at a reduction to handling and agility. Seemingly a bit more effective in a brawl."
```

File: code/modules/projectiles/guns/rifles.dm [hunk 5]
```diff
@@ -296,36 +297,40 @@
...
 //M41A PMC VARIANT
 
-/obj/item/weapon/gun/rifle/m41a1/elite
-	name = "\improper M41A2 battle rifle"
-	desc = "A refined and redesigned version of the tried and tested M41A1 Pulse Rifle. Given only to elite units."
-	icon_state = "m41a2"
-	item_state = "m41a2"
+/obj/item/weapon/gun/rifle/m412/elite
+	name = "\improper M412E battle rifle"
+	desc = "This M412 has seen severe modification, its been given a stock and a longer barrel with an intergrated barrel charger, and a red skull on the body for some reason.."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/rifles.dm:327]:
```suggestion
	desc = "An \"Elite\" modification of the M412 pulse rifle series, given to special operation units. It has been given a stock and a longer barrel with an integrated barrel charger, with a red skull stenciled on the body for some reason."
```

File: _maps/map_files/Pillar_of_Spring/TGS_Pillar_of_Spring.dmm [hunk 6]
```diff
@@ -757,8 +757,8 @@
 /obj/machinery/camera/autoname/mainship{
 	dir = 4
 	},
-/obj/item/weapon/gun/rifle/m41a1{
-	desc = "An outdated rifle for the TerraGov Marine Corps, carried by a few coporate mercenary groups, the M41A1 is a very rare sight in TerraGov systems. Uses 10x24mm caseless ammunition. Inscribed is Moth"
+/obj/item/weapon/gun/rifle/m412{
+	desc = "The M412 rifle is a Pulse Industries rifle, billed as a pulse rifle due to its use of electronic firing for faster velocity. A rather common sight in most systems. Uses 10x24mm caseless ammunition. Inscribed is a moth stencil and it looks worn."
```
Comment (Hughgent) [inline _maps/map_files/Pillar_of_Spring/TGS_Pillar_of_Spring.dmm:761]: There is a worn inscription of a moth on it.

Although faint, there is a moth inscribed on it.

Stenciled on the side is a worn inscription of a moth.

**Review by Hughgent — APPROVED** [review]
LGTM

**Comment by XSlayer300** [comment]
Back sprites when the rifle is both held and unwelded are inverted, it is supposed to be in the other way around.

(AK for reference on what SHOULD it look like)
image

**Comment by XSlayer300** [comment]
I feel like the rate of fire of the M41A should be buffed too, since it feels too slow to fire especially on full-auto.

**Comment by XSlayer300** [comment]
image

**Comment by XSlayer300** [comment]
Also, you can still attach muzzle attachments to the M412E, despite having an integrated barrel charger (and that includes attaching a barrel charger too). Suppressor and bayonet are my choices for the gun's attachments.

File: _maps/map_files/Pillar_of_Spring/TGS_Pillar_of_Spring.dmm (hunk 6 above)
Comment (XSlayer300) [reply _maps/map_files/Pillar_of_Spring/TGS_Pillar_of_Spring.dmm:761]: 1st one sounds better.

**Comment by Pariah919** [comment]
>I feel like the rate of fire and burst fire delay of the M41A should be buffed too, since it feels too slow to fire especially on full-auto.

I'm not touching the M41A in this PR other than minor lore edits.

>Misaligned GL.

Should be fixed.

>Also, you can still attach muzzle attachments to the M412E, despite having an integrated barrel charger (and that includes attaching a barrel charger too). Suppressor and bayonet are my choices for the gun's attachments.
All in all, I feel like this needs more work.

This gun is like used by PMTLs and that's basically in normal gameplay, but sure.

**Comment by Pariah919** [comment]
PR is effectively done if no other changes are requested.

**Comment by Hathkar** [comment]
Burst spits out bullets FAST, while automatic is quite slow in comparison.
Either the Automatic fire is too slow, or the burst fire is too fast.

**Comment by Pariah919** [comment]
>Burst spits out bullets FAST, while automatic is quite slow in comparison.
Either the Automatic fire is too slow, or the burst fire is too fast.

That's the HPR. I haven't touched it other than making it fire slightly faster.

**Review by Hughgent — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-51). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**